modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaHelidonCommonCodegen.java

```
@@ -34,4 +35,23 @@ public abstract class JavaHelidonCommonCodegen extends AbstractJavaCodegen
     static final String SERIALIZATION_LIBRARY_JACKSON = "jackson";
     static final String SERIALIZATION_LIBRARY_JSONB = "jsonb";
 
+    static final String DEFAULT_HELIDON_VERSION = "2.5.0";
+
+    private String helidonVersion = DEFAULT_HELIDON_VERSION;
+
+    @Override
+    public void processOpts() {
+        super.processOpts();
+
+        if (additionalProperties.containsKey(CodegenConstants.PARENT_VERSION)) {
+            helidonVersion = additionalProperties.get(CodegenConstants.PARENT_VERSION).toString();
+        }
+
+        if (additionalProperties.containsKey(CodegenConstants.HELIDON_VERSION)) {
+            helidonVersion = additionalProperties.get(CodegenConstants.HELIDON_VERSION).toString();
+        }
+
+        parentVersion = helidonVersion;
```

I am not sure about that. If these two values are not equal maybe it is better to let the user know that these two values must be equal because it can lead to ambiguity. Maybe it is better to throw an exception.

---

Conflict with @spericas previous review:

> If one is set to X the other must be set to X as well, and they should not both be set to different values.

---

I agree with assertion that these two values have to equal, but I do not think that we have to do the assignment operation. If you think that an exception is bad choice for this case we have to mention about these two values in our docs.

---

If only one of the two is set, we should set the other to the same value. If both are set to different values, I'm fine with an exception.

---

Fixed in latest commit.